org.scala-ide.sdt.core/src/org/scalaide/util/Utils.scala

```diff
@@ -46,7 +50,37 @@ object Utils extends HasLogger {
      */
     def asInstanceOfOpt[B: ClassTag]: Option[B] = obj match {
       case b: B => Some(b)
-      case _    => None
+      case _ => None
+    }
+  }
+
+  /**
+   * Scala wrapper on [[JdiInvocationSynchronizer]]. Remark: the actor thread can be killed in any moment
+   * by [[ActorSystem]] so the lock is not released then. To avoid it call [[jdiSynchronized.apply]] wrapped
+   * in [[Future]] like in example:
+   * {{{
+   *   def receive = {
+   *     case msg =>
+   *       val replyTo = sender
+   *       Future {
+   *         jdiSynchronized {
```

Comment:
Couldn't you move Future inside jdiSynchronized? This way user won't be able to make mistake and use it without Future:

```
object jdiSynchronized {
    def apply[T, R](code: => T)(fin: => R): Unit = {
        Future { JdiInvocationSynchronizer.instance.runSynchronized(callable(() => code)) } onComplete { fin }
    }
}
```